crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,147 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use tako::messages::gateway::CollectedOverview;
+
+use crate::dashboard::ui::terminal::DashboardFrame;
+use tui::layout::Rect;
+use tui::style::{Color, Modifier, Style};
+use tui::symbols;
+use tui::text::Span;
+use tui::widgets::{Axis, Block, Borders, Chart, Dataset};
+
+/// A new WorkerTimeState is created everytime there is a #worker change event
+#[derive(Clone)]
+struct WorkerStateAtTime {
+    /// The time at which the event being stored happened, wrt. the reference
+    pub at_time: u128,
+    pub event: TimeEventType,
+}
+
+#[derive(Clone)]
+pub enum TimeEventType {
+    /// Current number of workers.
+    WorkerCountUpdate(u32),
+    /// The average cpu utilization of the worker.
+    AvgCpuUsageUpdate(f32),
+}
+
+#[derive(Clone)]
+pub struct ClusterOverviewChart {
+    /// The state of the worker at different times.
+    timed_info_vec: Vec<WorkerStateAtTime>,
+    /// Used to set the min bound of the chart's x-axis
+    time_reference: u128,
+    /// Used to set the max bound of the chart's y-axis
+    max_observed_workers: u64,
+    /// Current number of workers connected
+    current_conn_workers: u64,
+}
+
+impl ClusterOverviewChart {
+    pub fn update(&mut self, overview: CollectedOverview) {
+        // Update the number of workers connected
+        let time = get_current_time();
+        let num_workers = overview.worker_overviews.len();
+        if num_workers > self.max_observed_workers as usize {
+            self.max_observed_workers = num_workers as u64;
+        }
+        self.timed_info_vec.push(WorkerStateAtTime {
+            at_time: time,
+            event: TimeEventType::WorkerCountUpdate(num_workers as u32),
+        });
+        // Add more points at the same time to smoothen the transition in plot
+        let interpolation =
+            interpolate_y_axis(self.current_conn_workers as f64, num_workers as f64, 30_f64);
```

Comment:
Maybe just remove this for the moment. It might look a bit weird, but the number of workers is a discrete value, so it makes sense that it will change in discrete steps.